pyramid/tests/test_scripts/test_pserve.py

```diff
@@ -77,6 +80,32 @@ def test_config_file_finds_watch_files(self):
             os.path.abspath(os.path.join(here, '*.py')),
         ])
 
+    def test_reload_call_hupper_with_correct_args(self):
+        from pyramid.scripts import pserve
+
+        class AttrDict(dict):
+            def __init__(self, *args, **kwargs):
+                super(AttrDict, self).__init__(*args, **kwargs)
+                self.__dict__ = self
+
+        def dummy_start_reloader(*args, **kwargs):
+            dummy_start_reloader.args = args
+            dummy_start_reloader.kwargs = kwargs
+
+        pserve.hupper = AttrDict(is_active=lambda: False,
+                                 start_reloader=dummy_start_reloader)
+
+        inst = self._makeOne('--reload', 'development.ini')
+        inst.run()
```

Comment:
I know we aren't using mock (sorry!) but if you're patching a module-scope value, you need to replace it after the test is over. For example:

```
orig_hupper = pserve.hupper
try:
    pserve.hupper = ...
    inst = ...
    inst.run()
finally:
    pserve.hupper = orig_hupper
```